## Support Vector Machines (SVM)

<blockquote><code>yasin inal

In [3]:
import sklearn as sk
from sklearn import datasets
from sklearn import svm

data = datasets.load_breast_cancer()

In [44]:
print(data.feature_names)
print(data.target_names)

x = data.data
y = data.target

x_train, x_test, y_train, y_test = sk.model_selection.train_test_split(x,y, test_size= 0.2)

['mean radius' 'mean texture' 'mean perimeter' 'mean area'
 'mean smoothness' 'mean compactness' 'mean concavity'
 'mean concave points' 'mean symmetry' 'mean fractal dimension'
 'radius error' 'texture error' 'perimeter error' 'area error'
 'smoothness error' 'compactness error' 'concavity error'
 'concave points error' 'symmetry error' 'fractal dimension error'
 'worst radius' 'worst texture' 'worst perimeter' 'worst area'
 'worst smoothness' 'worst compactness' 'worst concavity'
 'worst concave points' 'worst symmetry' 'worst fractal dimension']
['malignant' 'benign']


In [19]:
x_train, y_train

(array([[1.440e+01, 2.699e+01, 9.225e+01, ..., 5.563e-02, 2.345e-01,
         6.464e-02],
        [1.275e+01, 1.670e+01, 8.251e+01, ..., 8.045e-02, 3.071e-01,
         8.557e-02],
        [9.668e+00, 1.810e+01, 6.106e+01, ..., 2.500e-02, 3.057e-01,
         7.875e-02],
        ...,
        [1.205e+01, 2.272e+01, 7.875e+01, ..., 1.092e-01, 2.191e-01,
         9.349e-02],
        [1.742e+01, 2.556e+01, 1.145e+02, ..., 1.099e-01, 1.603e-01,
         6.818e-02],
        [1.624e+01, 1.877e+01, 1.088e+02, ..., 1.732e-01, 2.770e-01,
         1.063e-01]]),
 array([1, 1, 1, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 0, 0, 1,
        0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1,
        1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1,
        1, 0, 0, 1, 0, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0,
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0,
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 1, 1, 1, 0, 1,
 

## SVC Without Parameters

In [12]:
from sklearn import metrics

classes = ["malignant", "benign"]

model = svm.SVC()
model.fit(x_train, y_train)

y_pred = model.predict(x_test)

acc = metrics.accuracy_score(y_test, y_pred)

print(acc)

0.9298245614035088


<blockquote> Accuracy is <code> 0.92

## SVC With a 'Kernel' Parameter

In [18]:
model = svm.SVC(kernel="linear")
model.fit(x_train, y_train)

y_pred = model.predict(x_test)

acc = metrics.accuracy_score(y_test, y_pred)

print(acc)

0.9736842105263158


<blockquote> Accuracy is <code> 0.97

<blockquote>The accuracy went from 0.92 to 0.97 just with one parameter change.